src/ucs/datastruct/ptr_map.inl

```diff
@@ -131,6 +132,7 @@ ucs_ptr_map_extract(ucs_ptr_map_t *map, ucs_ptr_map_key_t key)
     void *value;
 
     if (ucs_likely(!(key & UCS_PTR_MAP_KEY_INDIRECT_FLAG))) {
+        ucs_assert(key != 0);
```

Comment:
why needed?

Reply:
@yosefe due to the discussion here - https://github.com/openucx/ucx/pull/5769/files#r503383561